Fix `VertxWebClientIT` condition assertion & correct Span operation name

michalvavrik opened the pull request:
### Summary

List of all traces were always retrieved as query parameter is in fact called [operation](https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/query_parser.go#L36) instead of `operationName`. Condition `thenCheckOperationNamesIsEqualTo` has been adjusted as previously it wasn't asserting that all checked operation names were equal to the expected one. `JsonPath` expression returned f.e. `[[chuck/pong, HTTP GET, trace/ping], [HTTP GET, trace/ping, chuck/pong], [chuck/pong, HTTP GET, trace/ping]]` mapped to `List<String>`, thus condition did assert that just one of the operation names contained operation name, not all of them. It's important to realize that traces filtered by operation name are returning traces with that operation name and their children traces (e.g. if you use rest client from Resource method), this is nicely visualized by Jaeger UI and thus expected behavior for Jaeger internal API we are using.

Expected operation name is prefixed with forward slash now as span name creation strategy has [recently changed](https://github.com/quarkusio/quarkus/pull/24017)

There are 2 commits intentionally as **only** `
Filter traces by operation name and correct VertxWebClientIT assertion` should be backported, but both all file changes must be part of this PR. Please **do not** backport `Make Span name consistent with http.route.name`.

Please select the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

michalvavrik commented:
Very same test (`VertxWebClientIT`) is failing in daily runs, which proves the test failure is not related. I can look on the test failure if I find time.

pjgg commented on http/vertx-web-client/src/test/java/io/quarkus/ts/http/vertx/webclient/VertxWebClientIT.java:
```diff
@@ -199,4 +214,52 @@ private String getServiceName() {
+        }
+
+        private String getParentSpanID() {
+            if (references != null) {
```
What about `if (!Objects.isNull(references))` ?

michalvavrik replied:
What about `nonNull`?

pjgg commented on http/vertx-web-client/src/test/java/io/quarkus/ts/http/vertx/webclient/VertxWebClientIT.java:
```diff
@@ -199,4 +214,52 @@ private String getServiceName() {
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static final class Span implements Comparable<Span> {
+
```
Maybe we could create a variable that represents a ROOT_PARENT, just for makes it more "easy to read", and then replace all nulls by this variable.

private static final String ROOT_PARENT = null;

pjgg commented on http/vertx-web-client/src/test/java/io/quarkus/ts/http/vertx/webclient/VertxWebClientIT.java:
```diff
@@ -199,4 +214,52 @@ private String getServiceName() {
+        private boolean isRootElement() {
+            return getParentSpanID() == null;
+        }
+
```
private boolean isRootElement() {
            return getParentSpanID() == ROOT_PARENT;
        }

pjgg commented on http/vertx-web-client/src/test/java/io/quarkus/ts/http/vertx/webclient/VertxWebClientIT.java:
```diff
@@ -199,4 +214,52 @@ private String getServiceName() {
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        private static final class References {
+            @JsonProperty
+            String spanID;
```
return ROOT_PARENT;

pjgg commented on http/vertx-web-client/src/test/java/io/quarkus/ts/http/vertx/webclient/VertxWebClientIT.java:
```diff
@@ -199,4 +214,52 @@ private String getServiceName() {
+
+        private String getParentSpanID() {
+            if (references != null) {
+                return references.stream().map(s -> s.spanID).findFirst().orElse(null);
```
return references.stream().map(s -> s.spanID).findFirst().orElse(ROOT_PARENT);

pjgg reviewed (changes requested):
There are some suggestions related to code changes on this PR (but over all looks good to me). 

And there on request change:

On module: `monitoring/opentelemetry` there are some requests to Jaeger API that are using `operationName` as a query param. Please, replace this query param and also be sure that this patch is not also required in these scenarios

michalvavrik commented:
> There are some suggestions related to code changes on this PR (but over all looks good to me).
> 
> And there on request change:
> 
> On module: `monitoring/opentelemetry` there are some requests to Jaeger API that are using `operationName` as a query param. Please, replace this query param and also be sure that this patch is not also required in these scenarios

I refactored "almost" accordingly to your suggestions, let see it, please. I have corrected query param in Opentelemetry stuff, but assertions was okay (passed because there are no other traces, so list of all = list of operation traces). Thank you for the review.

pjgg reviewed (approved):
LGTM. 

This PR depends on: https://github.com/quarkus-qe/quarkus-test-suite/pull/708
Once is merged, needs to be rebased and then will be ready to merge

michalvavrik commented:
This PR depends on #709 too.

michalvavrik commented:
Validation error has no connection to this PR and I can't see what I can do about it!

michalvavrik commented:
All checked passed and reviewer approved changes, thus merging.
